extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php

```diff
@@ -404,51 +402,58 @@ public function execute() {
 		/**
 		 * check if not additional domain was used (then we redirect anyway)
 		 */
-		$cond2 = $this->mAlternativeDomainUsed && ( $url['host'] != $this->mOldServerName );
+		$cond2 = $this->mAlternativeDomainUsed &&
+			( $url['host'] != WikiFactory::normalizeHost( $this->mOldServerName ) );
 
 		$redirectUrl = WikiFactory::getLocalEnvURL( $this->mCityUrl );
 		$shouldUseHttps = ( $wgEnableHTTPSForAnons || !empty( $_SERVER['HTTP_FASTLY_SSL'] ) ) &&
 			wfHttpsAllowedForURL( $redirectUrl ) &&
 			!empty( $_SERVER['HTTP_FASTLY_FF'] );	// don't redirect internal clients
 		$shouldUpgradeToHttps = $shouldUseHttps && empty( $_SERVER['HTTP_FASTLY_SSL'] );
 
-		if ( ( $cond1 || $cond2 || $shouldUpgradeToHttps ) &&  $wgWikiFactoryRedirectForAlternateDomains ) {
+		if ( $cond1 || $cond2 || $shouldUpgradeToHttps ) {
 			if ( $shouldUseHttps ) {
 				$redirectUrl = wfHttpToHttps( $redirectUrl );
 			}
-			$target = rtrim( $redirectUrl, '/' ) . '/' . $this->pathParams;
-
-			$queryParams = $_GET;
-			$localArticlePathClean = str_replace( '$1', '', $wgArticlePath );
-			if ( !empty( $localArticlePathClean ) &&
-				!empty( $queryParams['title'] ) &&
-				startsWith( $this->pathParams,  ltrim( $localArticlePathClean, '/' ) ) ) {
-				// skip the 'title' which is part of the $target, but append remaining parameters
-				unset( $queryParams['title'] );
+			if ( isset( $this->parsedUrl['path'] ) ) {
+				$redirectUrl .= $this->parsedUrl['path'];
 			}
-
-			if ( !empty( $queryParams ) ) {
-				$target .= '?' . http_build_query( $queryParams );
+			if ( isset( $this->parsedUrl['query'] ) ) {
+				$redirectUrl .= '?' . $this->parsedUrl['query'];
+			}
+			if ( isset( $this->parsedUrl['fragment'] ) ) {
+				$redirectUrl .= '#' . $this->parsedUrl['fragment'];
 			}
 
-			header( "X-Redirected-By-WF: NotPrimary" );
-			header( 'Vary: Cookie,Accept-Encoding' );
-
-			global $wgCookiePrefix;
-			$hasAuthCookie = !empty( $_COOKIE[\Wikia\Service\User\Auth\CookieHelper::ACCESS_TOKEN_COOKIE_NAME] ) ||
-				!empty( $_COOKIE[session_name()] ) ||
-				!empty( $_COOKIE["{$wgCookiePrefix}Token"] ) ||
-				!empty( $_COOKIE["{$wgCookiePrefix}UserID"] );
+			$redirectedBy = [];
+			if ( $shouldUpgradeToHttps ) $redirectedBy[] = 'AnonsHTTPSUpgrade';
+			if ( $cond1 ) $redirectedBy[] = 'NotPrimary';
+			if ( $cond2 ) $redirectedBy[] = 'AlternativeDomain';
+			$redirectedBy = join( ' ', $redirectedBy );
 
-			if ( $hasAuthCookie ) {
-				header( 'Cache-Control: private, must-revalidate, max-age=0' );
+			global $wgWFLRedirect;
```

Comment:
Note: replace with a hook call, name is something like `WikiFactoryLoaderRedirect`

Reply:
or not. we cannot easily register a hook listener inside index.php, as the `Hooks` class and `$wgHooks` do not exist yet